src/app/public/modules/text-editor/services/text-editor.service.ts

```diff
@@ -0,0 +1,501 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  Observable,
+  Subject
+} from 'rxjs';
+
+import {
+  STYLE_STATE_DEFAULTS
+} from '../defaults/style-state-defaults';
+
+import {
+  EditorCommand
+} from '../types/editor-command';
+
+import {
+  EditorSetting
+} from '../types/editor-setting';
+
+import {
+  SkyTextEditorStyleState
+} from '../types/style-state';
+
+import {
+  UrlModalResult
+} from '../url-modal/text-editor-url-modal-result';
+
+import {
+  UrlTarget
+} from '../url-modal/text-editor-url-target';
+
+import {
+  SkyTextSelectionManagementService
+} from './text-selection-management.service';
+
+/**
+ * @internal
+ */
+@Injectable()
+export class SkyTextEditorService {
```

Comment:
There is a lot in this file as a whole that we tend to send to an adapter.

Reply:
Agreed... I was hoping to defer this a bit after beta, but I noticed you had a couple other comments below around an adapter service, so I figured why not just go for it... 🤷 

We now have 4 services:

1. `SkyTextEditorService` - maintains the master dictionary of active editors. Has a few calls to get the listeners by id.
2. `SkyTextEditorAdapterService` - anything that touches the Web API (which is the meat and potatoes of this particular module since its a wysiwyg html editor 😄 )
3. `SkyTextEditorSelectionService` - Isolated methods for accessing the document's selection.
4. `SkyTextSanitizationService` - contains a single `sanitize()` method.

All the above are singletons, as they can be implemented from multiple components and need to maintain multiple instances of text editors. How do you feel about this?

Reply:
Works for me. Thanks for knocking this out as I know it was probably a chore :)